libtelemetry/src/main/java/com/mapbox/services/android/telemetry/MapboxTelemetry.java

```diff
@@ -277,4 +278,29 @@ public void onServiceDisconnected(ComponentName className) {
       serviceBound = false;
     }
   };
+
+  private boolean checkLocationPermission() {
+    if (PermissionsManager.areLocationPermissionsGranted(context)) {
+      return true;
+    } else {
+      permissionBackoff();
+      return false;
+    }
+  }
+
+  private void permissionBackoff() {
+    Handler handler = new Handler();
+    ExponentialBackoff counter = new ExponentialBackoff();
+
+    PermissionCheckRunnable permissionCheckRunnable = new PermissionCheckRunnable(context, this);
+
+    long nextWaitTime = counter.nextBackOffMillis();
```

Comment:
No need to calculate `nextWaitTime` and calling `handler.postDelayed(permissionCheckRunnable, nextWaitTime)` because of ☝️
What about calling `permissionCheckRunnable.run()` instead?